daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/VariableMap1.scala

```diff
@@ -233,21 +240,26 @@ final class VariableBox(initialVMap: VariableMap) {
  * The DPath implementation must be made to implement the
  * no-set-after-default-value-has-been-read behavior. This requires that reading the variables causes a state transition.
  *
- * The implementation of the VariabeMap uses ArrayBuffers essentially as a
- * stack, as they allow for easy serialization unlike the custom MStack classes
- * we use elsewhere. Scala's mutable Stack is deprecated in 2.12
+ * The VariableMap implementation essentially uses Seq as a stack, as they allow for easy
+ * serialization unlike the custom MStack classes we use elsewhere. Note that Scala's mutable
```

Comment:
Does this suggest a revisit (not in this change set) of MStack generally? 

The point of MStack was to avoid allocating objects when pushing value classes to/from stacks. If that's not worth it because we encounter overhead duplicating the MStacks....  Maybe they should be reimplemented in a simpler way or just go away entirely?

Reply:
Yeah, it's definitely possible we should take a look at MStack. I didn't see it pop up in my profiling, but I didn't do a lot and just focuses and the really big stuff like this variable map. I hope to do some more extensive profiling in the next release so I'll make sure to keep any eye out.